e2e/utils/test_cases.js

```diff
@@ -55,7 +55,8 @@ function getTM4JTestCases(report) {
             };
         }).
         reduce((acc, item) => {
-            const key = item.title.split(' ')[0].split('_')[0];
+            // Extract the key to exactly match with "MM-T[0-9]+"
+            const key = item.title.match(/(MM-T\d+)/)[0];
```

Comment:
Actual fix is here. Other changes were not necessary but good to change for standard formatting.